src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/searchemail/GenericSearchEmailService.java

```diff
@@ -0,0 +1,34 @@
+package de.dhbw.karlsruhe.students.mailflow.core.application.email.searchemail;
+
+import de.dhbw.karlsruhe.students.mailflow.core.application.email.searchemail.content.SearchEmailUseCase;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.Email;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.Mailbox;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.MailboxRepository;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.enums.MailboxType;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.exceptions.MailboxLoadingException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.exceptions.MailboxSavingException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Address;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class GenericSearchEmailService implements SearchEmailUseCase {
+
+  private final MailboxRepository mailboxRepository;
+
+  protected GenericSearchEmailService(MailboxRepository mailboxRepository) {
+    this.mailboxRepository = mailboxRepository;
+  }
```

Comment:
```suggestion
  private final ProvideEmailUseCase provideEmailUseCase;

  protected GenericSearchEmailService(ProvideEmailUseCase provideEmailUseCase) {
    this.provideEmailUseCase= provideEmailUseCase;
  }
```
Theoretically you could also use this service here. In this way you dont have to iterate through each mailbox but you can use the provideEmailUseCase.provideEmails() method